frontend/pages/hosts/ManageHostsPage/ManageHostsPage.tsx

```diff
@@ -585,14 +633,22 @@ const ManageHostsPage = ({
     //     queryParams: omit(queryParams, ["software_id"]),
     //   })
     // );
+    // TODO: If we work with directly with the location descriptor object from react-router,
+    // it is easy to just strip out the software_id and keep the rest of the path and query params.
+    // For example:
+    // router.replace({
+    //   pathname: location.pathname,
+    //   query: omit(queryParams, "software_id"),
+    // });
```

Comment:
Same, could this be documented somewhere else? Really appreciate you keeping an eye out for areas to improve later, and I want to make sure it doesn't get lost. Also helps to have outside the code where the teams can discuss.